core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTerms.java

```diff
@@ -67,6 +67,11 @@ public Object getKey() {
         @Override
         public Number getKeyAsNumber() {
             // this method is needed for scripted numeric aggs
+            try {
+                return Long.parseLong(termBytes.utf8ToString());
+            } catch (NumberFormatException ignored) {
+                //failed to be parsed as a Long, it will be parsed as a Double
```

Comment:
This comment is still not explaining why we are doing this. It's again an English translation of the code. May I suggest going back to the original issue and reading why we are making this change? That is what we want to say here, the why.